Application/EdFi.Ods.Common/Infrastructure/Configuration/NHibernateAutofacObjectsFactory.cs

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Linq;
+using Autofac;
+using NHibernate.Bytecode;
+
+namespace EdFi.Ods.Common.Infrastructure.Configuration
+{
+    public class NHibernateAutofacObjectsFactory : IObjectsFactory
+    {
+        private readonly IComponentContext _container;
+
+        public NHibernateAutofacObjectsFactory(ILifetimeScope container)
+        {
+            _container = container;
+        }
+
+        public object CreateInstance(Type type)
+        {
+            return _container.ResolveOptional(type) ?? Activator.CreateInstance(type);
```

Comment:
Resolving the type from the container includes an instance scope, however, new'ing it up will not include one.  Could this be a problem?